presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/ResourceGroupIdTemplate.java

```diff
@@ -53,7 +57,7 @@ public static ResourceGroupIdTemplate fromSegments(List<ResourceGroupNameTemplat
         return new ResourceGroupIdTemplate(String.join(".", segments.stream().map(ResourceGroupNameTemplate::toString).collect(Collectors.toList())));
     }
 
-    public ResourceGroupId expandTemplate(VariableMap context)
+    public SelectionContext<VariableMap> expandTemplate(VariableMap context)
```

Comment:
It seems like we're changing the semantics of this method.  It's not really just expanding the template into a resource group id, it's giving you an entire selection context.  I would keep this method unchanged but make it private, and make a new method which calls this method to get the id, and also does the below (and it's public).

Reply:
I will change it.